test/e2e/specs/editor/various/splitting-merging.spec.js

```diff
@@ -255,6 +255,26 @@ test.describe( 'splitting and merging blocks (@firefox, @webkit)', () => {
 		expect( content ).toBe( `` );
 	} );
 
+	test( 'should not merge heading with empty paragraph block on backspace', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+		await page.keyboard.type( 'Heading' );
+		await pageUtils.pressKeys( 'ArrowLeft', { times: 7 } );
+		await page.keyboard.press( 'Backspace' );
```

Comment:
After pressing backspace, it's good to type another character (e.g. "b) to test that the selection ended up in the right place. In this case it should result in a heading with "ba".

Reply:
Sorted!